internal/db/schema/migrations/oss/postgres/87/01_remove_session_connection_state.up.sql

```diff
@@ -0,0 +1,220 @@
+-- Copyright (c) HashiCorp, Inc.
+-- SPDX-License-Identifier: BUSL-1.1
+
+begin;
+
+  -- Remove the session_connection_state table and any related triggers
+  drop trigger update_connection_state_on_closed_reason on session_connection;
+  drop function update_connection_state_on_closed_reason();
+
+  drop trigger insert_session_connection_state on session_connection_state;
+  drop function insert_session_connection_state();
+
+  drop trigger update_session_state_on_termination_reason on session;
+  drop function update_session_state_on_termination_reason();
+
+  drop trigger insert_new_connection_state on session_connection;
+  drop function insert_new_connection_state();
+
+  drop trigger immutable_columns on session_connection_state;
+
+  drop trigger wh_insert_session_connection_state on session_connection_state;
+  drop function wh_insert_session_connection_state();
+
+  drop trigger wh_insert_session_connection on session_connection;
+  drop function wh_insert_session_connection();
+
+  drop table session_connection_state;
+  drop table session_connection_state_enm;
+
+  --  If the connected_time_range is null, it means the connection is authorized but not connected.
+  --  If the upper value of connected_time_range is > now() (upper range is infinity) then the state is connected.
+  --  If the upper value of connected_time_range is <= now() then the connection is closed.
+  alter table session_connection
+    add column connected_time_range tstzrange;
```

Comment:
Why not add the column and migrate the data from `session_connection_state`?

Reply:
Do you think it's worth migrating? I'm assuming that the control plane would go down for maintenance, and if we regularly remove terminated sessions from the db it seems like we should be fine to drop that table because sessions should be in a terminated state.

Reply:
I think we should migrate the data. We only delete the sessions after they have been terminated for an hour. So once the controllers are back up there will be an hour before the previous sessions are deleted, and I think we should report an accurate connection state for that time period.

Reply:
I've added a migration; unfortunately I can't add a migration test because of the code changes, but I've manually tested it.